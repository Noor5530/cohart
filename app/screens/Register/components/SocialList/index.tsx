import React from "react";
import { View, Text, TouchableOpacity, Image, Linking } from "react-native";

import Instagram from "assets/instagram.png";
import Twitter from "assets/twitter.png";
import Facebook from "assets/facebook.png";
import Cohart from "assets/cohart-social.png";
import Discord from "assets/discord.png";
import ArrowDown from "assets/Arrow_Down.png";

import styles from "./style";

const SocialList: React.FC = () => {
  const handleOpenInstagram = () => {
    Linking.canOpenURL("instagram://user?username=cohart_beta").then(
      (supported) => {
        if (supported) {
          Linking.openURL("instagram://user?username=cohart_beta");
        } else {
          Linking.openURL("https://www.instagram.com/cohart_beta/");
        }
      }
    );
  };

  const handleOpenTwitter = () => {
    Linking.openURL("https://twitter.com/cohart_co");
  };

  const handleOpenFacebook = () => {
    Linking.openURL("https://www.facebook.com/HelloCohart");
  };

  const handleOpenCohartWeb = () => {
    Linking.openURL("https://www.cohart.co");
  };

  const handleOpenDiscord = () => {
    Linking.openURL("https://discord.gg/9kpBJvkv");
  };

  return (
    <View style={{ flex: 1, backgroundColor: "white" }}>
      <View
        style={{
          width: 45,
          height: 7,
          backgroundColor: "rgba(0, 0, 0, 0.4)",
          borderRadius: 12,
          alignSelf: "center",
          marginTop: 10,
        }}
      />
      <View style={styles.socialContainer}>
        <TouchableOpacity onPress={handleOpenInstagram} style={styles.rowItem}>
          <Image source={Instagram} />
          <Text style={styles.socialText}>@cohart_beta</Text>
        </TouchableOpacity>
        <TouchableOpacity onPress={handleOpenTwitter} style={styles.rowItem}>
          <Image source={Twitter} />
          <Text style={styles.socialText}>@cohart_co</Text>
        </TouchableOpacity>
        <TouchableOpacity onPress={handleOpenFacebook} style={styles.rowItem}>
          <Image source={Facebook} />
          <Text style={styles.socialText}>/HelloCohart</Text>
        </TouchableOpacity>
        <TouchableOpacity onPress={handleOpenCohartWeb} style={styles.rowItem}>
          <Image source={Cohart} />
          <Text style={styles.socialText}>Cohart</Text>
        </TouchableOpacity>
        <TouchableOpacity onPress={handleOpenDiscord} style={styles.rowItem}>
          <Image source={Discord} />
          <Text style={styles.socialText}>Cohart</Text>
        </TouchableOpacity>
      </View>
      <Text style={styles.socialSwipeText}>swipe down to keep discovering</Text>
      <View style={styles.arrowWrapper}>
        <Image source={ArrowDown} />
      </View>
    </View>
  );
};

export default SocialList;
