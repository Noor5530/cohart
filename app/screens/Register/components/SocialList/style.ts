import { StyleSheet } from "react-native";
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from "react-native-responsive-screen";

import { appFonts } from "components/text";

const styles = StyleSheet.create({
  buttonTextWithIcon: {
    marginRight: wp("1"),
  },
  socialContainer: {
    paddingTop: hp("28"),
    paddingLeft: wp("7"),
  },
  rowItem: {
    flexDirection: "row",
    alignItems: "center",
    marginBottom: wp("4"),
  },
  socialText: {
    marginLeft: wp("5"),
    fontSize: 16,
    fontFamily: appFonts.InterRegular,
  },
  socialSwipeText: {
    textTransform: "uppercase",
    fontSize: 18,
    fontFamily: appFonts.InterSemiBold,
    marginTop: hp("2"),
    alignSelf: "center",
  },
  arrowWrapper: {
    alignSelf: "center",
    marginTop: hp("3"),
  },
});

export default styles;
