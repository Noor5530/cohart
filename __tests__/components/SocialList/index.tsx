import "react-native";
import React from "react";
import SocialList from "../../../app/screens/Register/components/SocialList";
import renderer from "react-test-renderer";

jest.useFakeTimers();

test("renders correctly", () => {
  const tree = renderer.create(<SocialList />).toJSON();

  expect(tree).toMatchSnapshot();
});
