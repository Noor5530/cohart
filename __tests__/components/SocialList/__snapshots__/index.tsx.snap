// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.4)",
        "borderRadius": 12,
        "height": 7,
        "marginTop": 10,
        "width": 45,
      }
    }
  />
  <View
    style={
      Object {
        "paddingLeft": undefined,
        "paddingTop": undefined,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": undefined,
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../app/assets/instagram.png",
          }
        }
      />
      <Text
        style={
          Object {
            "fontFamily": "Inter-Regular",
            "fontSize": 16,
            "marginLeft": undefined,
          }
        }
      >
        @cohart_beta
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": undefined,
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../app/assets/twitter.png",
          }
        }
      />
      <Text
        style={
          Object {
            "fontFamily": "Inter-Regular",
            "fontSize": 16,
            "marginLeft": undefined,
          }
        }
      >
        @cohart_co
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": undefined,
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../app/assets/facebook.png",
          }
        }
      />
      <Text
        style={
          Object {
            "fontFamily": "Inter-Regular",
            "fontSize": 16,
            "marginLeft": undefined,
          }
        }
      >
        /HelloCohart
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": undefined,
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../app/assets/cohart-social.png",
          }
        }
      />
      <Text
        style={
          Object {
            "fontFamily": "Inter-Regular",
            "fontSize": 16,
            "marginLeft": undefined,
          }
        }
      >
        Cohart
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": undefined,
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../app/assets/discord.png",
          }
        }
      />
      <Text
        style={
          Object {
            "fontFamily": "Inter-Regular",
            "fontSize": 16,
            "marginLeft": undefined,
          }
        }
      >
        Cohart
      </Text>
    </View>
  </View>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "fontFamily": "Inter-SemiBold",
        "fontSize": 18,
        "marginTop": undefined,
        "textTransform": "uppercase",
      }
    }
  >
    swipe down to keep discovering
  </Text>
  <View
    style={
      Object {
        "alignSelf": "center",
        "marginTop": undefined,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../app/assets/Arrow_Down.png",
        }
      }
    />
  </View>
</View>
`;
